src/components/LoansListItem.tsx

```diff
@@ -32,51 +44,23 @@ const LoanName = styled.h3`
   line-height: 1.2;
   margin-bottom: 10px;
 `;
-
-const Button = styled.button`
-  width: 150px;
-  height: 50px;
-  border-radius: 3px;
-  border: none;
-  cursor: pointer;
-  font-weight: 600;
-  font-size: 24px;
-
-  color: gray;
-  background: #ffff99;
-  text-transform: uppercase;
-  transition: all 150ms linear;
-  &:hover,
-  &:active {
-    background: #ffff00;
-    color: #0099cc;
-  }
-`;
-
-// "title": "Voluptate et sed tempora qui quisquam.",
-//       "tranche": "A",
-//       "available": "11,959",
-//       "annualised_return": "8.60",
-//       "term_remaining": "864000",
-//       "ltv": "48.80",
-//       "amount": "85,754"
-
-interface Loan {
-  id: String;
-  title: String;
-  available: String;
-  annualised_return: String;
-  term_remaining: String;
-  ltv: String;
-  amount: String;
+export interface Loan {
+  id: string;
+  title: string;
+  available: string;
+  annualised_return: string;
+  term_remaining: string;
+  ltv: string;
+  amount: string;
 }
 interface Props {
   loan: Loan;
-  setModalOpen: (arg: boolean) => void;
-  setCurrentLoan: (loan: object) => void;
+  invested: { [key: string]: boolean };
+  setModalOpen: (modalOpen: boolean) => void;
+  setCurrentLoan: (loan: Loan) => void;
 }
 
-const LoansListItem: React.FC<Props> = ({ loan, setModalOpen, setCurrentLoan }) => {
+const LoansListItem: React.FC<Props> = ({ loan, invested, setModalOpen, setCurrentLoan }) => {
   const onModalOpen = () => {
```

Comment:
there is no need to have 2 callbacks: setCurrentLoan and setModalOpen. Replace it with 1: onLoanInvest()